src/benchmark/file_io_read_micro_benchmark.cpp

```diff
@@ -14,9 +16,11 @@ class FileIOMicroReadBenchmarkFixture : public MicroBenchmarkBasicFixture {
  public:
   uint64_t control_sum = uint64_t{0};
   std::vector<uint32_t> numbers;
+  std::vector<uint32_t> random_indices;
 
   void SetUp(::benchmark::State& state) override {
     // TODO: Make setup/teardown global per file size to improve benchmark speed
+    // TODO: Generate vector for each loop in state and not only once per state
```

Comment:
I will do this on Monday :)

Reply:
Explanation: Currently we generate the numbers randomly for each state, so in a for(_:state) loop we always have the same numbers. We do clear the cache currently, but instead we should pull ths setup into each benchmark. Furthermore, this code duplication should then be refactored in a setup_sequential and setup_randomized function.

Reply:
I was not able to do this, @tjjordan can you have a look at this on monday?